### DISCUSSION TITLE
How to completely silence deprecations, esp during tests.

### DISCUSSION THREAD

**Discussion by tacman** [symfony/symfony#48218] [Q&A] [answered]
How do I completely silence deprecation messages?  During this stage of development, they're not helping, and in fact they're distracting me from important messages.

Furthermore, in PHP Storm when running tests they're impossible to read, they're this giant red blob telling me that my vendors aren't yet current.  So what -- I don't want to know about this right now.

Of course, I've added SYMFONY_DEPRECATIONS_HELPER=weak to .env.test.  And to phpunit.xml, in both the php section and the listener:

```xml
    <php>
        <env name="SYMFONY_DEPRECATIONS_HELPER" value="max[self]=0" />
        <ini name="display_errors" value="1" />
        <ini name="error_reporting" value="-1" />
        <server name="APP_ENV" value="test" force="true" />
        <server name="SHELL_VERBOSITY" value="-1" />
        <server name="SYMFONY_PHPUNIT_REMOVE" value="" />
        <server name="SYMFONY_PHPUNIT_VERSION" value="9.5" />
    </php>
    <listeners>
        <listener class="Symfony\Bridge\PhpUnit\SymfonyTestsListener">
            <arguments>
                <array>
                    <!-- set this option to 0 to disable the DebugClassLoader integration -->
                    <element key="debug-class-loader"><integer>0</integer></element>
                </array>
            </arguments>
        </listener>
    </listeners>
```

But nothing I do can silence their screams.  

![image](https://user-images.githubusercontent.com/619585/202040339-20ee81d3-8516-42be-8776-b54685ae5519.png)

And since the colors are something like "light grey on red",  I always have to do a double-take --it's red, is it important?  What does it say?  If it's important, why is so hard to read.  If it's a deprecation, why does it have to scream in red?

I'm using Symfony 6.2, so some of the warnings are even from Symfony, which of course is fine.  I just don't want to see them, so that when red flashes in my tests, I know to look at them.

**Comment by MatTheCat** [comment, marked as answer]
Did you try the `verbose`, `quiet` or even `disabled` options? See https://symfony.com/doc/current/components/phpunit_bridge.html#disabling-the-verbose-output

**Reply by tacman** [reply]
Thanks!  None of these worked in phpunit.xml

```xml
    <php>
<!--        <env name="SYMFONY_DEPRECATIONS_HELPER" value="verbose=0" />-->
<!--        <env name="SYMFONY_DEPRECATIONS_HELPER" value="disabled=1" />-->
<!--        <env name="SYMFONY_DEPRECATIONS_HELPER" value="max[self]=0" />-->
```

BUT in .env.test, this worked

    SYMFONY_DEPRECATIONS_HELPER=disabled=1

**Reply by MatTheCat** [reply]
Note that PHPUnit also provide a `server` tag and a `force` attribute. Don’t know if it would have solved anything but I remember I had to fiddle with these a long time ago.
